jvm/src/main/java/com/microsoft/bond/IBonded.java

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+import com.microsoft.bond.protocol.ProtocolWriter;
+
+import java.io.IOException;
+
+public interface IBonded<T extends BondSerializable> {
+    <W extends ProtocolWriter> void Serialize(W writer) throws IOException;
```

Comment:
This is based on the structure of the C# version. I have a sneaking feeling it'll be back, but changed for now.